compiler_gym/datasets/datasets.py

```diff
@@ -0,0 +1,480 @@
+        """
+        added = False
+        # Use an attribute check rather than isinstance() to allow mocking the
+        # Dataset class in testing.
+        if hasattr(dataset, "name"):
+            added = self.add(dataset)
+            dataset_name = dataset.name
+        else:
+            dataset_name = resolve_uri_protocol(dataset)
+        activated = self.activate(dataset_name)
+        installed = self.install(dataset_name)
+        return added or activated or installed
+
+    def benchmarks(self) -> Iterable[Benchmark]:
+        """Enumerate the (possibly infinite) benchmarks lazily.
+
+        Benchmarks order is consistent across runs. Benchmarks from each dataset
+        are returned in order. The order of :meth:`benchmarks()
+        <compiler_gym.datasets.Datasets.benchmarks>` and :meth:`benchmark_uris()
+        <compiler_gym.datasets.Datasets.benchmark_uris>` is the same.
+
+        :return: An iterable sequence of :class:`Benchmark
+            <compiler_gym.datasets.Benchmark>` instances.
+        """
+        for dataset in self.datasets():
+            yield from dataset.benchmarks()
+
+    def benchmark_uris(self) -> Iterable[str]:
+        """Enumerate the (possibly infinite) benchmark URIs.
+
+        Benchmark URI order is consistent across runs. URIs from each dataset
+        are returned in order. The order of :meth:`benchmarks()
+        <compiler_gym.datasets.Datasets.benchmarks>` and :meth:`benchmark_uris()
+        <compiler_gym.datasets.Datasets.benchmark_uris>` is the same.
+
+        :return: An iterable sequence of benchmark URI strings.
+        """
+        for dataset in self.datasets():
+            yield from dataset.benchmark_uris()
+
+    def benchmark(self, uri: Optional[str] = None) -> Benchmark:
+        """Select a benchmark.
+
+        If a benchmark URI is given, the corresponding :class:`Benchamrk
+        <compiler_gym.datasets.Benchmark>` is returned, regardless of whether
+        the containing dataset is active, hidden, or installed.
+
+        If no URI is given, a benchmark is selected randomly. Fist, a dataset is
+        selected uniformly randomly from the set of available datasets. Then a
+        benchmark is selected randomly from the chosen dataset.
+
+        Calling :code:`benchmark()` will yield benchmarks from all available
+        datasets with equal probability, regardless of how many benchmarks are
+        in each dataset. Given a pool of available datasets of differing sizes,
+        smaller datasets will be overrepresented and large datasets will be
+        underrepresented.
+
+        Use :meth:`seed() <compiler_gym.datasets.Dataset.seed>` to force a
+        reproducible order for randomly selected benchmarks.
+
+        :param uri: The URI of the benchmark to return. If :code:`None`, select
+            a benchmark randomly using :code:`self.random`.
+
+        :return: A :class:`Benchamrk <compiler_gym.datasets.Benchmark>`
```

Comment:
s/Benchamrk/Benchmark/

Reply:
Thanks! Turns out I made that typo all over the place :) #199